src/ObjectUtils/LinuxMap.cpp

```diff
@@ -89,28 +89,156 @@ ErrorMessageOr<std::vector<ModuleInfo>> ReadModules(int32_t pid) {
   return ParseMaps(proc_maps_data);
 }
 
+namespace {
+
+// Because of the requirements on the arguments of mmap, if the .text section of a Portable
+// Executable has an offset in the file (PointerToRawData, multiple of FileAlignment) that is not
+// congruent to the offset of that section when loaded into memory (VirtualAddress, multiple of
+// SectionAlignment) modulo the page size, Wine cannot create a file-backed mapping for the .text
+// section, and resorts to copying into an anonymous mapping. This means that, for PE binaries with
+// this property, we cannot simply associate an executable mapping to the corresponding file using
+// the path in the mapping.
+// However, we can make an educated guess. The path of the PE will at least appear in the read-only
+// mapping that corresponds to the beginning of the file, which contains the headers (because the
+// offset in the file is zero and the address chosen for this mapping should always be multiple of
+// the page size). If the executable file mapping for the .text section is not present, we consider
+// the anonymous executable mappings after the first file mapping for this PE: if the offset and
+// size of such a mapping are compatible with the address range where the .text section would be
+// loaded based on the header for the section (in particular, VirtualAddress and VirtualSize), we
+// can be quite sure that this is the mapping we are looking for.
+// This class contains logic to help ParseMaps with this detection mechanism.
+// Note that we assume that a PE (or an ELF file) only has one .text section and one executable
+// mapping: this is what we observed and is what we support.
```

Comment:
As this class implements a kind of state machine, adding a comment about the intended usage would be nice.

Reply:
I added a paragraph. Maybe have a quick look.